### ISSUE TITLE
overlapping regions in bcftools annotate

### ISSUE THREAD

**Issue by dmckean** [samtools/bcftools#943]
Annotating a VCF with a BED that contains overlapping regions only seems to output the first region in the BED. Consider `bcftools annotate -a regions.bed.gz -c CHROM,FROM,TO,ABC -h ABC_Tag.txt input.vcf.gz`

regions.bed(.gz)
```
1	1141500	1141520	A
1	1141510	1141600	B
```

Output
```
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO
1	1141506	.	G	A	.	PASS	ABC=A
1	1141517	.	C	A	.	PASS	ABC=A
1	1141556	.	G	C	.	PASS	ABC=B
1	1141570	.	G	C	.	PASS	ABC=B
1	1141643	.	T	C	.	PASS	.
```

If the INFO header has `Number=.`, any ideas for annotating both? I've also messing around in bedtools to form a split version of the bed instead. Something like this, but smart enough to handle more than two overlapping regions.
```
1	1141500	1141510	A
1	1141510	1141520	A,B
1	1141520	1141600	B
```

**Comment by pd3** [comment]
This is now addressed by af6f0c9. With this commit, you should be able to achieve what you need by running `bcftools annotate -a regions.bed.gz -c CHROM,FROM,TO,ABC -l ABC:unique -h ABC_Tag.txt input.vcf.gz`. 

The new `-l` option can be used as documented in the man page https://github.com/samtools/bcftools/blob/af6f0c964ac40c1ba5b71811b456b84ef0ebee43/doc/bcftools.txt#L366-L374

So far this works only for the special case of strings.

**Comment by dmarkie** [comment]
I have been attempting to usethe experimental enhancement described in #943 to annotate variants sensibly when they fall within overlapping intervals in the annotation source file, and have encountered two issues. I have limited annotations to only Type=String, as this feature is not implemented yet for annotations of Type=Float and Type=Integer, and I have been using only the append option, as again most of the other options are not yet implemented. Here I am attempting to annotate with five different columns from the file, and using bcftools version 1.9-204-g6244ac9+htslib-1.9-247-g6eacc77.
`bcftools annotate -a source.txt.gz -c CHROM,FROM,TO,A,B,C,D,E -h header.hdr -l A:append,B:append,C:append,D:append input.vcf`

1. When the vcf file contains any variants that fall outside an interval defined in the annotation source (ie should simply not be annotated), then the operation crashes with a segmentation fault. When the vcf is edited to only include variants within those intervals, then annotation completes.

2. When annotation does complete, although it appears that annotations have been correctly appended on the first line, subsequent lines contain the first letter of the appropriate annotation but the rest of the string is derived from the last annotation of the first line. Below I have shown the "appended" annotations from the INFO fields of the first two lines in the vcf - they should both be identical as they are in the same interval overlap.

`A=B3GALT6,B3GALT6;B=EHLERS-DANLOS_SYNDROME,SPONDYLOEPIMETAPHYSEAL_DYSPLASIA_WITH_JOINT_LAXITY_TYPE_1;C=confirmed,confirmed;D=biallelic,biallelic;E=loss_of_function,loss_of_function`

`A=Boss_of_function,loss_of_function;B=Eoss_of_function,loss_of_function;C=coss_of_function,loss_of_function;D=boss_of_function,loss_of_function;E=loss_of_function,loss_of_function`
